aws/resource_aws_glue_trigger.go

```diff
@@ -0,0 +1,457 @@
+										}, false),
+									},
+									"state": {
+										Type:     schema.TypeString,
+										Required: true,
+										ValidateFunc: validation.StringInSlice([]string{
+											glue.JobRunStateFailed,
+											glue.JobRunStateStopped,
+											glue.JobRunStateSucceeded,
+											glue.JobRunStateTimeout,
+										}, false),
+									},
+								},
+							},
+						},
+						"logical": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  glue.LogicalAnd,
+							ValidateFunc: validation.StringInSlice([]string{
+								glue.LogicalAnd,
+								glue.LogicalAny,
+							}, false),
+						},
+					},
+				},
+			},
+			"schedule": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					glue.TriggerTypeConditional,
+					glue.TriggerTypeOnDemand,
+					glue.TriggerTypeScheduled,
+				}, false),
+			},
+		},
+	}
+}
+
+func resourceAwsGlueTriggerCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).glueconn
+	name := d.Get("name").(string)
+	triggerType := d.Get("type").(string)
+
+	input := &glue.CreateTriggerInput{
+		Actions: expandGlueActions(d.Get("actions").([]interface{})),
+		Name:    aws.String(name),
+		Type:    aws.String(triggerType),
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		input.Description = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("predicate"); ok {
+		input.Predicate = expandGluePredicate(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("schedule"); ok {
+		input.Schedule = aws.String(v.(string))
+	}
+
+	if d.Get("enabled").(bool) && triggerType != glue.TriggerTypeOnDemand {
+		input.StartOnCreation = aws.Bool(true)
+	}
+
+	log.Printf("[DEBUG] Creating Glue Trigger: %s", input)
+	_, err := conn.CreateTrigger(input)
+	if err != nil {
+		return fmt.Errorf("error creating Glue Trigger (%s): %s", name, err)
+	}
+
+	d.SetId(name)
+
+	log.Printf("[DEBUG] Waiting for Glue Trigger (%s) to create", d.Id())
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{
+			glue.TriggerStateActivating,
+			glue.TriggerStateCreating,
+		},
+		Target: []string{
+			glue.TriggerStateActivated,
+			glue.TriggerStateCreated,
+		},
+		Refresh: resourceAwsGlueTriggerRefreshFunc(conn, d.Id()),
+		Timeout: d.Timeout(schema.TimeoutCreate),
+	}
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("error waiting for Glue Trigger (%s) to create", d.Id())
+	}
+
+	return resourceAwsGlueTriggerRead(d, meta)
+}
+
+func resourceAwsGlueTriggerRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).glueconn
+
+	input := &glue.GetTriggerInput{
+		Name: aws.String(d.Id()),
+	}
+
+	log.Printf("[DEBUG] Reading Glue Trigger: %s", input)
+	output, err := conn.GetTrigger(input)
+	if err != nil {
+		if isAWSErr(err, glue.ErrCodeEntityNotFoundException, "") {
```

Comment:
Luckily, we don't get crashes from the SDK in these situations. The SDK will bubble up some sort of networking error (I believe always `RequestError`), which is related to handling like this I just added: https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/config.go#L354-L372

As for a generic "is not found" function -- all the AWS service APIs return something different in this regard (status codes and error codes, some services have error codes for each different resource as well), so unfortunately it would probably be more trouble than its worth to try and create a single function to handle that.